src/families/cardano/js-getTransactionStatus.ts

```diff
@@ -0,0 +1,68 @@
+import { BigNumber } from "bignumber.js";
+import {
+  NotEnoughBalance,
+  RecipientRequired,
+  FeeNotLoaded,
+  InvalidAddress,
+  AmountRequired,
+} from "@ledgerhq/errors";
+import type { Account, TransactionStatus } from "../../types";
+import type { Transaction } from "./types";
+import { isValidAddress } from "./logic";
+import { utils as TyphonUtils } from "@stricahq/typhonjs";
+import { getCurrentCardanoPreloadData } from "./preload";
+import { CardanoMinAmountError } from "./errors";
+
+const getTransactionStatus = async (
+  a: Account,
+  t: Transaction
+): Promise<TransactionStatus> => {
+  const errors: Record<string, Error> = {};
+  const warnings: Record<string, Error> = {};
+  const useAllAmount = !!t.useAllAmount;
+
+  const estimatedFees = t.fees || new BigNumber(0);
+  const amount = t.amount;
+  const totalSpent = new BigNumber(amount).plus(estimatedFees);
+  const tokensToSend = []; //TODO: read from transaction
+
+  const cardanoPreloadedData = getCurrentCardanoPreloadData();
+
+  if (!t.fees) {
+    errors.fees = new FeeNotLoaded();
+  }
+
+  if (!t.recipient) {
+    errors.recipient = new RecipientRequired();
+  } else if (!isValidAddress(t.recipient)) {
+    errors.recipient = new InvalidAddress();
+  }
+
+  const minTransactionAmount = TyphonUtils.calculateMinUtxoAmount(
+    tokensToSend,
+    new BigNumber(cardanoPreloadedData.protocolParams.lovelacePerUtxoWord),
+    false
+  );
+
+  if (!amount.gt(0)) {
+    errors.amount = useAllAmount
+      ? new NotEnoughBalance()
+      : new AmountRequired();
+  } else if (amount.lt(minTransactionAmount)) {
+    errors.amount = new CardanoMinAmountError(
+      `Minimum ${minTransactionAmount.div(1e6)} ADA required`
```

Comment:
Error messages are defined in live-desktop and live-mobile, e.g:
- https://github.com/LedgerHQ/ledger-live-desktop/blob/615022c001c7858a1c1d2f9d00a716ecd6a41029/static/i18n/en/app.json#L3957-L3959
- https://github.com/LedgerHQ/ledger-live-mobile/blob/3978dfa2cf49e97c1a459ea0943afdb09814b495/src/locales/en/common.json#L396-L398

If you need custom errors than you'll need to open PRs on live-desktop and live-mobile. 
Here you should only provide the data to be used in the message.
From the example in the links you can see that you can provide the `currencyName` param to the `InvalidAddress` error above.